Sources/protoc-gen-swift/MessageStorageDecision.swift

```diff
@@ -0,0 +1,190 @@
+// Sources/protoc-gen-swift/MessageGenerator.swift - Per-message logic
+//
+// Copyright (c) 2014 - 2016 Apple Inc. and the project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See LICENSE.txt for license information:
+// https://github.com/apple/swift-protobuf/blob/master/LICENSE.txt
+//
+// -----------------------------------------------------------------------------
+
+import SwiftProtobufPluginLibrary
+
+// The file attempts to isolate the decisions around when to use heap based
```

Comment:
I love this. Having this pulled out into its own self-contained unit will make it a lot easier to reason about changes to the algorithm in the future.

Reply:
Thanks, I figured if this gets more work in the future it would be clearer this way.